connection/TypeDBSessionImpl.java

```diff
@@ -161,6 +161,10 @@ public void close() {
         }
     }
 
+    public void closed(TypeDBTransaction.Extended typeDBTransaction) {
+        transactions.remove(typeDBTransaction);
+    }
```

Comment:
I think this is okay to do even with the `transactions.forEach(TypeDBTransaction.Extended::close);` on L152 in `close()`, based on the similar construction in `SessionService` in typedb core.

Reply:
agree - the locking is used to not open transactions when the session is concurrently closed. This operation should always be fine though.

Reply:
I think this can not be `public`?

Reply:
You're right, this can be package access.